src/util/strencodings.h

```diff
@@ -67,6 +68,66 @@ std::vector<Byte> ParseHex(std::string_view hex_str)
 {
     return TryParseHex<Byte>(hex_str).value_or(std::vector<Byte>{});
 }
+
+/** consteval version of HexDigit() without the lookup table. */
+consteval uint8_t ConstevalHexDigit(const char c)
+{
+    if (c >= '0' && c <= '9') return c - '0';
+    if (c >= 'a' && c <= 'f') return c - 'a' + 0xA;
+
+    throw "Only lowercase hex digits are allowed, for consistency";
+}
+
+struct ConstevalHexLiteral {
+    const std::string_view inner;
+    template <size_t N>
+    consteval ConstevalHexLiteral(const char (&hex_str)[N]) : inner{hex_str} { Validate(inner); }
```

Comment:
Since it's a `consteval` ctor, I _think_ taking a `const char*` parameter should behave the same but avoid having to instantiate for each diferently sized `hex_str` input? But I'm still easily confused by how `consteval` behaves.

```suggestion
   consteval ConstevalHexLiteral(const char* hex_str) : inner{hex_str} { Validate(inner); }
```

Reply:
That's strange, I don't see how a string literal would need a conversion when there's a `const char*` ctor? It seems to work fine for [`ConstevalStringLiteral`](https://github.com/bitcoin/bitcoin/blob/2f7d9aec4d049701fccfc029f44934d187467432/src/util/translation.h#L72) too, so pardon my insistence but I think perhaps this compiler error might be because of another issue? Would you mind trying this diff on 734ac5a9002493013c0f8afe763f751ac99f89c8?

<details>
<summary>git diff on 734ac5a900</summary>

```diff
diff --git a/src/util/strencodings.h b/src/util/strencodings.h
index fbd116c77c..63f46396d3 100644
--- a/src/util/strencodings.h
+++ b/src/util/strencodings.h
@@ -80,8 +80,7 @@ consteval uint8_t ConstevalHexDigit(const char c)
 
 struct ConstevalHexLiteral {
     const std::string_view inner;
-    template <size_t N>
-    consteval ConstevalHexLiteral(const char (&hex_str)[N]) : ConstevalHexLiteral{std::string_view{hex_str}} {}
+    consteval ConstevalHexLiteral(const char* hex_str) : ConstevalHexLiteral{std::string_view{hex_str}} {}
     consteval ConstevalHexLiteral(const std::string_view hex_str) : inner{hex_str} { Validate(inner); }
     consteval ConstevalHexLiteral(std::nullptr_t) = delete;
 

```
</details>

Reply:
Thank you for being persistent! Must have had other local changes interacting badly with it when I tried it. Got it working with only your diff, so if we end up resurrecting `ConstevalHexLiteral` I'll go with your version.